### PR TITLE
created answer.js action and reducer

### LINKED ISSUE #121: Answer create action + reducer
<h1> Redux action for creating an answer </h1>

Resolves #117 

<h3> Fixes </h3>

- [x] Fix fetch function to send proper answer data
- [x] Make answer reducers properly update answer in store
- [x] Temporary answers not working? (_View doesn't update until after we get a fetch response_)
- [ ] React-controlled `RangeOption` should still update even if no dependent option was hit

<h3> Integrate into questionnaire </h3>

- [x] Make OptionsContainer dispatch an action to update answer in store
- [ ] Retry component to submit answer again when fetch request fails

<h3> Actions file </h3>

- [x]  Create Answers actions file

- [x]  Create `current_fetching` answer action
    - type: `IS_FETCHING` and any other parameters necessary for reducer

- [x] Create `update_answer` answer action
    - type: `UPDATE_ANSWER` and answer object to set as params

- [x] Create `done_fetching` answer action
    - inputs dispatch and serialized answer object
    - type: `DONE_FETCHING` and any other parameters necessary for reducer. 

<h3> Reducer file </h3>

- [x] Create Answers reducer file

- [x] Contains a case for `UPDATE_ANSWER`
    - The answer object should be added to the redux store state under the correct building object inside the answers object.

- [x] Contains a case for `IS_FETCHING`
    - when this action is dispatched, redux store state should have a key, value `{ fetching_answer_status: true }`

- [x] Contains a case for `DONE_FETCHING`
    - redux store state should have key `fetching_answer_status` set to `false`

<h3> Constants file </h3>

- [x] Add necessary constants for action type to constants file

<h3> Action file </h3>

- [x] edit_answer action (`PATCH` instead of `POST`)
- [x] create_answer action
    - Inputs dispatch and options for any necessary arguments (i.e. answer parameters)
    - pseudo:

```
createAnswer(dispatch, answer_params={}) {
     dispatch currently_fetching(set_fetching: true)
     await fetch( url='/api/answers/create', body=answer_params).then(
          function(response) { 
               dispatch update_answer(answer_params)
               dispatch done_fetching(set_fetching: false) 
          }
      )
}
```

### PR DESCRIPTION
**Pull request by JustinMi**
Resolves #121 

**Update Nov 18**
- Just started

**Update Nov 28**
- finished implementation
- Created an answer reducer and answer action
- there is currently implementation for answer creation, but no implementation for answer deletion

### CODE REVIEW

File: app/javascript/packs/actions/buildings.js [hunk 1]
```diff
@@ -81,26 +107,65 @@ export async function updateBuilding(id, updatedBuilding, dispatch) {
...
 
-    dispatch(saveResponse(UPDATE_BUILDING, id, response.data));
+    dispatch(saveBuilding(UPDATE_BUILDING, id, response.data));
   } catch (error) {
     dispatch(saveError(UPDATE_BUILDING, id, error));
   };
 }
 
-function saveResponse(type, id, response) {
+  /////////////////////////////////////////////////////////////////
```
Comment (HacksForHugs) [inline app/javascript/packs/actions/buildings.js]: You can actually have the answer actions exported in their own answers file in actions folder.

File: app/javascript/packs/reducers/buildings.js [hunk 2]
```diff
@@ -78,14 +87,102 @@ function saveBuilding(state, action) {
...
     case EDIT_BUILDING: return editBuilding(state, action);
     case REMOVE_BUILDING: return removeBuilding(state, action);
     case CREATE_BUILDING: return saveBuilding(state, action);
     case UPDATE_BUILDING: return saveBuilding(state, action);
+
+    // Answers
+    case CREATE_ANSWER: return createAnswer(state, action);
+    case UPDATE_ANSWER: return updateAnswer(state, action);
+    case REMOVE_ANSWER: return removeAnswer(state, action);
+    case SAVE_ANSWER: return saveAnswer(state, action);
```
Comment (HacksForHugs) [inline app/javascript/packs/reducers/buildings.js]: Remove this one and move all change the saveAnswer to updateAnswer. this follows naming convention for this type of action.

File: app/javascript/packs/constants/index.js [hunk 3]
```diff
@@ -1,5 +1,13 @@
 export const LOAD_INITIAL_STATE = 'LOAD_INITIAL_STATE';
 
+// Answer ActionTypes
+export const FETCH_ANSWER = 'FETCH_ANSWER';
+export const ADD_ANSWER = 'ADD_ANSWER';
+export const CREATE_ANSWER = 'CREATE_ANSWER';
+export const UPDATE_ANSWER = 'UPDATE_ANSWER';
+export const REMOVE_ANSWER = 'REMOVE_ANSWER';
+export const SAVE_ANSWER = 'SAVE_ANSWER';
```
Comment (HacksForHugs) [inline app/javascript/packs/constants/index.js]: remove this and move all logic for save answer to update answer for naming conventions

**Comment by ethanlee16** [comment]
@kevintli - feel free to submit this branch as a new PR when complete! Requested changes from Helen will be here for your review.

**Pull request closed without merging**